app/controllers/cards_controller.rb

```diff
@@ -4,9 +4,49 @@ def index
   end
 
   def show
-    @card = Cards.find(params[:id])
+    @card = Card.find(params[:id])
+  end
+
+  def edit
+    @card = Card.find(params[:id])
   end
 
   def new
+    @card = Card.new
+  end
+
+  def update
+     @card = Card.find(params[:id])
+
+    if @card.update(card_params)
+      redirect_to @card
+    else
+      render 'edit'
+    end
+
+  end
+
+  def create
+    @card = Card.new(card_params)
+
+    if @card.save 
+      redirect_to @card
+    else
+      render 'new'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.